src/BasisPotentials/basis_potentials.jl

```diff
@@ -0,0 +1,67 @@
+################################################################################
+# InteratomicPotentials API implementations for basis potentials
+################################################################################
+
+# Common basis potential functions for computing energy, forces, and virials
+
+function compute_global_descriptors(
+    B::Vector{Vector{T}},
+    bp::BasisPotential
+) where T<: Real
+    return sum(B)
+end
+
+function potential_energy(
+    s::AbstractSystem,
+    bp::BasisPotential;
+    T = Float64
+)
+    B = compute_local_descriptors(s, bp.basis, T = T)
```

Comment:
You'll have to update these function calls if you get rid of the default casting behavior that was added in